src/main/groovy/wooga/gradle/version/internal/release/opinion/Strategies.groovy

```diff
@@ -248,23 +275,12 @@ final class Strategies {
          * increment the count of the nearest any and append it to the so far inferred pre-release
          * component. Otherwise append 1 to the so far inferred pre-release component.
          */
-        static final PartialSemVerStrategy COUNT_INCREMENTED = closure { state ->
-            def nearest = state.nearestVersion
-            def currentPreIdents = state.inferredPreRelease ? state.inferredPreRelease.split('\\.') as List : []
-            if (nearest.any == nearest.normal || nearest.any.normalVersion != state.inferredNormal) {
-                currentPreIdents << '1'
-            } else {
-                def nearestPreIdents = nearest.any.preReleaseVersion.split('\\.')
-                if (nearestPreIdents.size() <= currentPreIdents.size()) {
-                    currentPreIdents << '1'
-                } else if (currentPreIdents == nearestPreIdents[0..(currentPreIdents.size() - 1)]) {
-                    def count = parseIntOrZero(nearestPreIdents[currentPreIdents.size()])
-                    currentPreIdents << Integer.toString(count + 1)
-                } else {
-                    currentPreIdents << '1'
-                }
+        static final PartialSemVerStrategy COUNT_INCREMENTED = countIncremented()
+
+        static final PartialSemVerStrategy countIncremented(String separator = ".", int countPadding = 0) {
```

Comment:
I see that you hadded now multiple helper function that return a closure to be used for state calcualtion.
Why not moving the seperator and countPadding into the `SemVerStrategyState`? I don't know if there are cases where we need to mix and match?

Reply:
There are cases where we need to divert from those defaults, see PreReleasePartials.java line 16. Also, I tried to avoid putting more state on `SemverStrategyState` as whatever is inside of it can be changed by any strategy.

Reply:
Ok then.